app/models/concerns/placement_query.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module PlacementQuery
+  extend ActiveSupport::Concern
+
+  def trainees_to_be_placed
+    previous_cycle = AcademicCycle.previous
+
+    trainees
```

Comment:
Just wondering about the rationale for delegating more of the work to ruby vs the database? Wouldn't having a pure sql query be more efficient? I also thought this was meant to be status independent, eg a provider could bulk add placements for trainees in any state (not entirely sure but assuming)

Reply:
I've re-worked the query to avoid looping